fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -681,7 +682,43 @@ private ParameterDAO makeParameterDAO(Connection connection) throws FHIRPersiste
                         }
                     } else {
                         unsortedResultsList = resourceDao.search(query);
-                        resources = this.convertResourceDTOList(unsortedResultsList, resourceType, elements);
+                        List<com.ibm.fhir.persistence.jdbc.dto.Resource> matchResultList = unsortedResultsList;
+                        List<com.ibm.fhir.persistence.jdbc.dto.Resource> includeResultList = new ArrayList<>();
+
+                        // Check if _include or _revinclude search. If so, remove duplicates from 'include' resources
+                        // (duplicates of both 'match' and 'include' resources) and make sure _elements processing
+                        // is not done for 'include' resources.
+                        if (searchContext.hasIncludeParameters() || searchContext.hasRevIncludeParameters()) {
+                            // Calculate 'match' results count
+                            int pageSize = searchContext.getPageSize();
+                            int offset = (searchContext.getPageNumber() - 1) * pageSize;
+                            int matchResultCount = pageSize;
+                            if (searchResultCount < offset + pageSize) {
+                                matchResultCount = searchResultCount - offset;
+                            }
+
+                            // Split results list into 'match' list and 'include' list
+                            if (unsortedResultsList.size() > matchResultCount) {
+                                matchResultList = unsortedResultsList.subList(0, matchResultCount);
+
+                                // Remove duplicates from 'include' list
+                                Set<Long> resultIds = new HashSet<>();
+                                for (com.ibm.fhir.persistence.jdbc.dto.Resource resource : matchResultList) {
+                                    resultIds.add(resource.getId());
+                                }
+                                for (int i=matchResultCount; i<unsortedResultsList.size(); ++i) {
+                                    com.ibm.fhir.persistence.jdbc.dto.Resource resource = unsortedResultsList.get(i);
+                                    if (!resultIds.contains(resource.getId())) {
+                                        resultIds.add(resource.getId());
+                                        includeResultList.add(resource);
```

Comment:
I've stared at this logic for a few moments. 

Could you use a LinkedHashMap (preserve order) and use a computeIfNotPresent? 

@punktilious thoughts on optimizations?

Reply:
not quite sure how I would take advantage of that here...I have to build the includeResultList to pass to the convertResourceDTOList() on line 721